### PR TITLE
Adding documentation for restart policy

### LINKED ISSUE #1353: missing description regarding restart (not restart_policy)
### Problem description

I was looking for
restart options but could not find any entries especially the option that existed none, on-failure and always can't be found anymore. also they are not marked as deprecated.

### Problem location

- I couldn't find the information I wanted. I expected to find it near the following URL: https://docs.docker.com/compose/compose-file/

### Project version(s) affected
compose 1.10

### Suggestions for a fix

I remember this info was in older docs, but I can't find them anymore.

### PR DESCRIPTION
**Pull request by surajnarwade**
### Proposed changes

Adding `restart` documenation in detail

### Related issues (optional)

Resolves #1353

### CODE REVIEW

File: compose/compose-file/compose-file-v2.md [hunk 1]
```diff
@@ -902,7 +902,17 @@ then read-write will be used.
 >     - container_name
 >     - container_name:rw
 
-### cpu\_shares, cpu\_quota, cpuset, domainname, hostname, ipc, mac\_address, mem\_limit, memswap\_limit, mem\_swappiness, mem\_reservation, oom_score_adj, privileged, read\_only, restart, shm\_size, stdin\_open, tty, user, working\_dir
```
Comment (johndmulhausen) [inline compose/compose-file/compose-file-v2.md]: I don't think we want to remove this heading. And we should probably change this in V3's reference too.

Comment (surajnarwade) [inline compose/compose-file/compose-file-v2.md]: I didn't removed that, just added new `restart` heading

**Comment by londoncalling** [comment]
@surajnarwade thanks for adding this to v.3 file, and yes I see that you didn't actually remove that long heading so that's cool. Can you just do a copyedit to improve the intro description for restart in both places/files as follows?

no is the default restart policy, and it will not restart a container under any circumstance. When always is specified, the container always restarts. The on-failure policy restarts a container if the exit code indicates an on-failure error.

**Comment by surajnarwade** [comment]
@londoncalling done with changes

**Comment by londoncalling** [comment]
@surajnarwade looks good, I'll merge this and close the associated issue as soon as the Netlify preview builds for a final check. Thanks!

File: compose/compose-file/index.md [hunk 2]
```diff
@@ -1209,13 +1209,20 @@ volumes:
     source: ./static
     target: /opt/app/static
 ```
+### restart
```
Comment (londoncalling) [inline compose/compose-file/index.md]: @surajnarwade you need another return before the heading for the restart command or the markdown will show (as it is now on Netlify: https://deploy-preview-2912--docker-docs.netlify.com/compose/compose-file/#long-syntax-2). This only seems to be a problem on v3 (`index.md`) file. It seems to be done correctly on the v2 file.

Comment (surajnarwade) [inline compose/compose-file/index.md]: @londoncalling thanks :)

**Comment by surajnarwade** [comment]
@londoncalling , what is error now ?

**Comment by londoncalling** [comment]
@surajnarwade it's nothing you did. We are having problems with Netlify. We're working on getting it working again. I'll ping you on this one when we've got those builds working again. Sorry!

**Comment by londoncalling** [comment]
@surajnarwade I just kicked Netlify, let's see if it works this time.

**Comment by londoncalling** [comment]
LGTM, @surajnarwade , merging

**Pull request merged**